enterprise/server/auth_service/auth_service.go

```diff
@@ -3,18 +3,30 @@ package auth_service
 import (
 	"context"
 
+	"github.com/buildbuddy-io/buildbuddy/server/interfaces"
 	"github.com/buildbuddy-io/buildbuddy/server/real_environment"
+	"github.com/buildbuddy-io/buildbuddy/server/util/claims"
 
 	authpb "github.com/buildbuddy-io/buildbuddy/proto/auth"
 )
 
 type AuthService struct {
+	authenticator interfaces.GRPCAuthenticator
 }
 
 func Register(env *real_environment.RealEnv) {
-	env.SetAuthService(AuthService{})
+	env.SetAuthService(AuthService{authenticator: env.GetAuthenticator()})
 }
 
 func (a AuthService) Authenticate(ctx context.Context, req *authpb.AuthenticateRequest) (*authpb.AuthenticateResponse, error) {
-	return &authpb.AuthenticateResponse{}, nil
+	u, err := a.authenticator.AuthenticateGRPCRequest(ctx)
```

Comment:
AuthenticatedGRPCRequest automatically puts the JWT into the context. If you use the JWT from the context you wouldn't need to mess with having expiration on UserInfo.

Reply:
Done